### PR TITLE
Upgrade Smarty to 3.1.47

### PR DESCRIPTION
**Pull request by mflasquin**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | 1.7.8.x
| Description?      | Upgrade Smarty to 3.1.47 : <br>- Fixes illegal characters warning in math. <br>- Prevent PHP injection through malicious block name or include file name. This addresses https://github.com/advisories/GHSA-634x-pc3q-cf4c. <br>- Fixed replace modifier by converting encoding if needed. <br>- Fixed second param of unescape. <br>- Applied appropriate javascript and html escaping in mailto plugin to counter injection attacks <br>- Fixed use of rand() without a parameter in math function <br>- Fixed unselected year/month/day not working in html_select_date 
| Type?             | improvement
| Category?         | CO
| BC breaks?        |  no
| Deprecations?     | no
| Fixed ticket?     | 
| Related PRs       |
| How to test?      | CI is green
| Possible impacts? |

### CODE REVIEW

File: composer.lock [hunk 1]
```diff
@@ -11370,5 +11368,5 @@
     "platform-overrides": {
         "php": "7.1.3"
     },
-    "plugin-api-version": "2.2.0"
+    "plugin-api-version": "1.1.0"
```
Comment (matks) [inline composer.lock]: Woups, please do not commit this line ;)

**Review by matks — CHANGES_REQUESTED** [review]
Thank you for this PR 👍 there is a little problem in composer.lock

**Review by matks — APPROVED** [review]

**Review by kpodemski — APPROVED** [review]

**Comment by kpodemski** [comment]
@PrestaShop/qa-team 

Hello QA Team! I think automatic tests should do the job for this one, right?

**Comment by Robin-Fischer-PS** [comment]
@kpodemski Yes, indeed, only automated tests :) Thanks for the ping !

**Comment by boubkerbribri** [comment]
Hello @PrestaShop/prestashop-core-developers  
Since this is not a regression, is the PR targeting the correct branch ?
Thank you

**Comment by matks** [comment]
> Hello @PrestaShop/prestashop-core-developers Since this is not a regression, is the PR targeting the correct branch ? Thank you

@boubkerbribri It's for security 😉 the last version of smarty contains a vulnerability bugfix and must be included in 1.7.8

**Comment by boubkerbribri** [comment]
> @boubkerbribri It's for security wink the last version of smarty contains a vulnerability bugfix and must be included in 1.7.8

Thanks, I'm taking it back

**Review by boubkerbribri — APPROVED** [review]
Tested with some UI tests : FO + some legacy pages.
=> QA :heavy_check_mark:

**Comment by matks** [comment]
Thank you @mflasquin

**Pull request merged**